dali/operators/util/normal_distribution_op.h

```diff
@@ -0,0 +1,152 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+#define DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+
+#include <random>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+const std::string kMean = "mean";      // NOLINT
+const std::string kStddev = "stddev";  // NOLINT
+const std::string kShape = "shape";    // NOLINT
+
+const int kNumOutputs = 1;
+auto kShapeDefaultValue = std::vector<int>{-1};
+
+}  // namespace detail
+
+template<typename Backend>
+class NormalDistribution : public Operator<Backend> {
+ public:
+  ~NormalDistribution() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(NormalDistribution);
+
+ protected:
+  explicit NormalDistribution(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          shape_(spec.GetArgument<std::remove_const_t<decltype(this->shape_)>>(detail::kShape)),
+          seed_(spec.GetArgument<std::remove_const_t<decltype(this->seed_)>>("seed")),
+          dtype_(spec.GetArgument<std::remove_const_t<decltype(this->dtype_)>>(
+                  arg_names::kDtype)) {}
+
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  void AcquireArguments(const workspace_t<CPUBackend> &ws) {
+    this->GetPerSampleArgument(mean_, detail::kMean, ws);
+    this->GetPerSampleArgument(stddev_, detail::kStddev, ws);
+  }
+
+
+  TensorListShape<> GetOutputShape(const workspace_t<CPUBackend> &ws) {
+    DALI_ENFORCE(!(spec_.NumRegularInput() == 1 && IsShapeArgumentProvided(spec_)),
+                 make_string("Incorrect operator invocation. "
+                             "The operator cannot be called with both Input and `shape` argument"));
+    if (spec_.NumRegularInput() == 1) {
+      single_value_in_output_ = false;
+      return ShapesFromInputTensorList(ws);
+    } else if (IsShapeArgumentProvided(spec_)) {
+      single_value_in_output_ = false;
+      return ShapesFromArgument(ws);
+    } else {
+      single_value_in_output_ = true;
+      return ShapeForDefaultConfig(ws);
+    }
+  }
+
+  using Operator<Backend>::spec_;
+  using Operator<Backend>::batch_size_;
+  std::vector<float> mean_, stddev_;
+  decltype(detail::kShapeDefaultValue) shape_;
+  const int64_t seed_;
+  const DALIDataType dtype_;
+
+  /**
+   * When this is true it means, that neither Input or Argument have been provided
+   * and the operator should generate one scalar per tensor in a batch (aka `CoinFlip` mode)
```

Comment:
would not mention CoinFlip here (see reason above)

Reply:
Removed